### PR TITLE
Skip reverse DNS lookups

### LINKED ISSUE #52: Connection timeout with nodejs app occasionally
I am using docker-compose to connect a nodejs app to a mysql database with the following `.yml` file:
```yml
db:
  image: mysql
  ports:
    - "3306:3306"
  environment:
    - MYSQL_ROOT_PASSWORD=password
    - MYSQL_DATABASE=sane
server:
  image: artificial/docker-sails:iojs-full-stable-pm2
  command: sails lift --verbose
  volumes:
    - server/:/server
  ports:
    - "1337:1337"
  links:
    - db
```

Often it works, but often I get a timeout and a warning `'2015-03-04 14:40:37 1 [Warning] IP address '172.17.0.15' could not be resolved: Temporary failure in name resolution'` I am not entirely sure if the issue is here, or in the app itself, since if I switch to the postgresql I have no issues as well.

Still trying to nail down where exactly the issue arises, this is more to keep track of it in case it is an issue with the image being slow or something.

### PR DESCRIPTION
**Pull request by yosifkit**
fixes #52

### CODE REVIEW

**Comment by yosifkit** [comment]
```console
$ ./generate-stackbrew-library.sh | sed -r 's/docker-library/infosiftr/' > mysql
$ ./bashbrew/bashbrew.sh build ../mysql/mysql
Cloning mysql (git://github.com/infosiftr/mysql) ...
Processing mysql:5.5.45 ...
Processing mysql:5.5 ...
Processing mysql:5.6.26 ...
Processing mysql:5.6 ...
Processing mysql:5 ...
Processing mysql:latest ...
Processing mysql:5.7.7-rc ...
Processing mysql:5.7.7 ...
Processing mysql:5.7 ...
$ ./bashbrew/bashbrew.sh list --uniq ../mysql/mysql | xargs ./test/run.sh
testing mysql:5.5.45
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'mysql-basics' [4/4]...passed
testing mysql:5.6.26
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'mysql-basics' [4/4]...passed
testing mysql:5.7.7-rc
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'mysql-basics' [4/4]...mysqld failed to accept connections in a reasonable amount of time!
ERROR 2003 (HY000): Can't connect to MySQL server on 'mysql' (111)
failed
```

File: 5.6/Dockerfile [hunk 1]
```diff
@@ -30,7 +30,10 @@ RUN { \
 	&& rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql
 
 # comment out a few problematic configuration values
-RUN sed -Ei 's/^(bind-address|log)/#&/' /etc/mysql/my.cnf
+# don't reverse lookup hostnames, they are usually another container
+RUN sed -Ei 's/^(bind-address|log)/#&/' /etc/mysql/my.cnf \
+	&& echo 'skip-host-cache\nskip-name-resolve' | awk '{ print } /^\[mysqld\]$/ && c == 0 { c = 1; system("cat") }' /etc/mysql/my.cnf > /tmp/my.cnf \
```
Comment (tianon) [inline 5.6/Dockerfile]: This regex condition could instead be `$1 == "[mysqld]"`.

**Comment by tianon** [comment]
I'm definitely a fan of skipping these by default for the image, especially since so many users will be behind at least Docker's NAT by default, so incoming connections won't resolve and we're just wasting precious time.

ping @ltangvald; what're your thoughts here?

**Comment by yosifkit** [comment]
This leaves the `my.cnf` having this diff from it's previous state:
```diff
 [mysqld]
+skip-host-cache
+skip-name-resolve
 user		= mysql
 pid-file	= /var/run/mysqld/mysqld.pid
 socket		= /var/run/mysqld/mysqld.sock
```

**Comment by shofetim** [comment]
:+1:

**Comment by tianon** [comment]
LGTM

**Comment by joshunger** [comment]
I believe this breaks permissions/grants in certain cases?  Can this be a setting?  Or, how do you override "skip-name-resolve" in a file in the conf.d directory?

**Comment by tianon** [comment]
Sorry for the delay, @joshunger; I would _imagine_ something like `skip-name-resolve = off` or `skip-name-resolve = false` should work (given that the config is supposed to be key-value pairs), but I haven't tested to verify that.  If that doesn't work, the only other thing I can think of (besides overwriting this provided file completely) is to do something like `RUN sed -i '/skip-name-resolve/d' ...` to remove the line.

**Pull request merged**